internal/querycoord/global_meta_broker.go

```diff
@@ -267,41 +269,121 @@ func (broker *globalMetaBroker) parseIndexInfo(ctx context.Context, segmentID Un
 	return nil
 }
 
-func (broker *globalMetaBroker) getIndexInfo(ctx context.Context, collectionID UniqueID, segmentID UniqueID, schema *schemapb.CollectionSchema) ([]*querypb.VecFieldIndexInfo, error) {
-	// TODO:: collection has multi vec field, and build index for every vec field, get indexInfo by fieldID
-	// Currently, each collection can only have one vector field
-	vecFieldIDs := funcutil.GetVecFieldIDs(schema)
-	if len(vecFieldIDs) != 1 {
-		err := fmt.Errorf("collection %d has multi vec field, num of vec fields = %d", collectionID, len(vecFieldIDs))
-		log.Error("get index info failed",
-			zap.Int64("collectionID", collectionID),
-			zap.Int64("segmentID", segmentID),
-			zap.Error(err))
-		return nil, err
-	}
-	indexInfo := &querypb.VecFieldIndexInfo{
-		FieldID: vecFieldIDs[0],
+// Better to let index params key appear in the file paths first.
+func (broker *globalMetaBroker) loadIndexExtraInfo(ctx context.Context, fieldPathInfo *indexpb.IndexFilePathInfo) (*extraIndexInfo, error) {
+	indexCodec := storage.NewIndexFileBinlogCodec()
+	for _, indexFilePath := range fieldPathInfo.IndexFilePaths {
+		// get index params when detecting indexParamPrefix
+		if path.Base(indexFilePath) == storage.IndexParamsKey {
+			content, err := broker.cm.MultiRead([]string{indexFilePath})
+			if err != nil {
+				return nil, err
+			}
+
+			if len(content) <= 0 {
```

Comment:
why not check len(content) == 1 directly?

Reply:
ok